agent/collector.go

```diff
@@ -270,12 +271,23 @@ func (l *Collector) postToAPI(endpoint config.APIEndpoint, checkPath string, bod
 	req.Header.Add("X-Dd-Hostname", l.cfg.HostName)
 	req.Header.Add("X-Dd-Processagentversion", Version)
 
+	// we create a context with timeout and attach it to the http request, to make sure that it doesn't get stuck on
+	// any circumstances
+	start := time.Now()
+	ctx, cancel := context.WithTimeout(context.Background(), HTTPTimeout)
```

Comment:
I think we want the timeout to be longer than the http client timeout to make sure the http request can finish. thoughts?